### PR TITLE
Fix null pointer deref in diff if no Upstream

### PR DESCRIPTION
**Pull request by Ralph7C2**
Running `kpt pkg diff` without an upstream set panics because we attempt
to dereference `kptFile.Upstream` without checking to see if it is nil. 

This commit adds a check and returns early if necessary.

### CODE REVIEW

File: internal/util/diff/diff.go
Comment (natasha41575) [inline internal/util/diff/diff.go:128]: nit: I suggest including the package path in the error message. 

```suggestion
		return errors.Errorf("package missing upstream in Kptfile at '%s', c.Path")
```

**Review by natasha41575 — COMMENTED** [review]
Thanks for the fix! I can reproduce the error on the main branch and I agree an error message is far better than panicking.

File: internal/util/diff/diff.go
Comment (natasha41575) [inline internal/util/diff/diff.go:128]: Agh sorry my bad

```suggestion
		return errors.Errorf("package missing upstream in Kptfile at '%s'", c.Path)
```

File: internal/util/diff/diff.go
Comment (Ralph7C2) [reply internal/util/diff/diff.go]: Good call, thanks!

File: internal/util/diff/diff.go
Comment (Ralph7C2) [reply internal/util/diff/diff.go]: :+1:

**Review by natasha41575 — APPROVED** [review]
LGTM (and built kpt from this PR to verify the fix). Will defer to @mortent or @droot (or another code owner) to give their final approval and merge.

**Comment by droot** [comment]
Awesome! Merging this in.. Thanks.

**Pull request merged**
